pkg/devspace/dependency/resolver.go

```diff
@@ -245,6 +246,16 @@ func (r *resolver) resolveDependency(ctx devspacecontext.Context, dependencyConf
 		return nil, err
 	}
 
+	if val, ok := dConfigWrapper.Variables()["DEVSPACE_ENV_FILE"]; ok {
```

Comment:
Not sure if that is a good way of doing this. The problem here is that the config is already resolved here and any variable that would get changed through the `DEVSPACE_ENV_FILE` wouldn't be reflected in the resolved config. While this probably fixes it for some use cases, I don't think we should go with just that as the variables need to be available before actually parsing the config.

Reply:
Yes, it needs to be re-worked.